packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -337,52 +324,108 @@ export class CdkToolkit {
 
       let elapsedDeployTime = 0;
       try {
-        const result = await this.props.deployments.deployStack({
-          stack,
-          deployName: stack.stackName,
-          roleArn: options.roleArn,
-          toolkitStackName: options.toolkitStackName,
-          reuseAssets: options.reuseAssets,
-          notificationArns,
-          tags,
-          execute: options.execute,
-          changeSetName: options.changeSetName,
-          deploymentMethod: options.deploymentMethod,
-          force: options.force,
-          parameters: Object.assign({}, parameterMap['*'], parameterMap[stack.stackName]),
-          usePreviousParameters: options.usePreviousParameters,
-          progress,
-          ci: options.ci,
-          rollback: options.rollback,
-          hotswap: options.hotswap,
-          hotswapPropertyOverrides: hotswapPropertyOverrides,
-          extraUserAgent: options.extraUserAgent,
-          assetParallelism: options.assetParallelism,
-          ignoreNoStacks: options.ignoreNoStacks,
-        });
+        let deployResult: RegularDeployStackResult | undefined;
 
-        const message = result.noOp
+        let rollback = options.rollback;
+        while (!deployResult) {
+          const r = await this.props.deployments.deployStack({
+            stack,
+            deployName: stack.stackName,
+            roleArn: options.roleArn,
+            toolkitStackName: options.toolkitStackName,
+            reuseAssets: options.reuseAssets,
+            notificationArns,
+            tags,
+            execute: options.execute,
+            changeSetName: options.changeSetName,
+            deploymentMethod: options.deploymentMethod,
+            force: options.force,
+            parameters: Object.assign({}, parameterMap['*'], parameterMap[stack.stackName]),
+            usePreviousParameters: options.usePreviousParameters,
+            progress,
+            ci: options.ci,
+            rollback,
+            hotswap: options.hotswap,
+            hotswapPropertyOverrides: hotswapPropertyOverrides,
+            extraUserAgent: options.extraUserAgent,
+            assetParallelism: options.assetParallelism,
+            ignoreNoStacks: options.ignoreNoStacks,
+          });
+
+          switch (r.type) {
```

Comment:
Indefinite `while` loop with a `switch` statement that doesn't have a `default` case. ❌ 

I'd suggest to implement this as a recursive function call for readability. Either way, we need a breaking default case and probably should only allow 2 attempt before failing.

Reply:
Recursive functions scare me more than while loops.

Reply:
`Recursive` vs `loop` is mainly about readability. 

You still need a way to prevent an infinite loop. In other places we limit this to max 10 iterations. 
You should also add a default case that breaks the loop.

Reply:
Yeah, I added both.